lib/game_box_web/live/upload_live.ex

```diff
@@ -20,53 +21,141 @@ defmodule GameBoxWeb.UploadLive do
+          id="upload-game-form"
+          for={@changeset}
+          phx-change="validate"
+          phx-submit="upload_game"
+        >
+          <.input field={{f, :title}} label="Title" />
+
+          <.input field={{f, :description}} label="Description" />
+
+          <.live_file_input upload={@uploads.game} />
+
+          <%= for err <- upload_errors(@uploads.game) do %>
+            <p class="alert alert-danger"><%= error_to_string(err) %></p>
+          <% end %>
+          <%= for entry <- @uploads.game.entries do %>
+            <article class="upload-entry">
+              <progress value={entry.progress} max="100"><%= entry.progress %>%</progress>
+
+              <button
+                type="button"
+                phx-click="cancel-game-upload"
+                phx-value-ref={entry.ref}
+                aria-label="cancel"
+              >
+                &times;
+              </button>
+
+              <%= for err <- upload_errors(@uploads.artwork, entry) do %>
+                <p class="alert alert-danger"><%= error_to_string(err) %></p>
+              <% end %>
+            </article>
+          <% end %>
+
+          <.live_file_input upload={@uploads.artwork} />
+
+          <%= for entry <- @uploads.artwork.entries do %>
+            <article class="upload-entry">
+              <figure>
+                <.live_img_preview entry={entry} />
+                <figcaption><%= entry.client_name %></figcaption>
+              </figure>
+
+              <progress value={entry.progress} max="100"><%= entry.progress %>%</progress>
+
+              <button
+                type="button"
+                phx-click="cancel-art-upload"
+                phx-value-ref={entry.ref}
+                aria-label="cancel"
+              >
+                &times;
+              </button>
+
+              <%= for err <- upload_errors(@uploads.artwork, entry) do %>
+                <p class="alert alert-danger"><%= error_to_string(err) %></p>
+              <% end %>
+            </article>
+          <% end %>
+          <%= for err <- upload_errors(@uploads.artwork) do %>
+            <p class="alert alert-danger"><%= error_to_string(err) %></p>
+          <% end %>
+          <:actions>
+            <.button type="submit" name="save">Save</.button>
+          </:actions>
+        </.simple_form>
+      </div>
       <div class="mt-8">
         <.h2>My Games</.h2>
-        <ul>
-          <li :for={game <- @games}>
-            <p><%= game.title %></p>
-          </li>
-        </ul>
+        <div class="grid grid-cols-3 gap-2">
+          <%= for game <- @games do %>
+            <div class="mb-4 border-2 border-white p4">
+              <.h2 class="text-center"><%= game.title %></.h2>
+              <img class="max-w-full h-auto" src={game.artwork} />
+            </div>
+          <% end %>
+        </div>
       </div>
     </div>
     """
   end
 
+  defp presign_upload(entry, %{assigns: %{uploads: uploads}} = socket) do
```

Comment:
It seems like a lot of this stuff  should live in our `SimpleS3Upload` module. I dont think the liveview should be responsible for knowing the s3 config or generating the s3 path url etc.

Reply:
hmmm you might be right